torch_geometric/data/graph_store.py

```diff
@@ -0,0 +1,126 @@
+from abc import abstractmethod
+from collections.abc import MutableMapping
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, Optional, Union
+
+from torch import LongTensor
+
+from torch_geometric.typing import EdgeTensorType, EdgeType
+from torch_geometric.utils.mixin import CastMixin
+
+
+class EdgeLayout(Enum):
+    COO = 'coo'
+    CSC = 'csc'
+    CSR = 'csr'
+
+
+@dataclass
+class EdgeAttr(CastMixin):
+    r"""Defines the attributes of an :obj:`GraphStore` edge."""
+
+    # The type of the edge
+    edge_type: Optional[Any]
```

Comment:
I'm biased against doing this since we don't need views just yet in `GraphStore`, and every use of `EdgeAttr` requires `edge_type` and `layout` to be set. I think we can avoid this complexity for the time being, and introduce it when necessary.